sale_commission_check_deposit/wizard/wizard_check_settle.py

```diff
@@ -0,0 +1,40 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+import json
+from datetime import date, timedelta
+
+from odoo import models
+
+
+class SaleCommissionMakeSettle(models.TransientModel):
+    _inherit = "sale.commission.make.settle"
+
+    def _get_agent_lines(self, agent, date_to_agent):
+        """
+        Exclude invoices with checks payment if at least safety days
+        haven't passed since expiration date set on check deposit.
+        """
+        agent_lines = super()._get_agent_lines(agent, date_to_agent)
+        for line in agent_lines.filtered(
+            lambda r: r.commission_id.invoice_state == "paid"
+            and r.invoice_id.invoice_payments_widget != "false"
+        ):
+            if self.get_move_ids(line.invoice_id.invoice_payments_widget):
+                agent_lines -= line
```

Comment:
not getting this one, can you elaborate?

Reply:
what about:

```
     move_lines = self.env["account.move.line"].search(
            [
                ("move_id", "in", account_move_ids),
                ("journal_id.is_check_journal", "=", True),
                ("check_deposit_id", "=", True),
            ])
     return any([x.date_maturity + timedelta(days=x.journal_id.safety_days) < date.today() for x in move_lines])
```

It check if you have at least 1 `move_line`, with `check_deposit_id`, having `maturity + safety < today` and return `true`. In this case you can remove line from agent.

Could this do the job?

Reply:
If returns at least 1 move_line it means that I need to settle that commission. But if it returns nothing I will remove it. I'll publish it soon. Thanks.

Reply:
I published the changes..